authutils/user.py

```diff
@@ -0,0 +1,155 @@
+import functools
+import json
+
+from addict import Dict
+from cached_property import cached_property
+from cdiserrors import AuthZError
+import flask
+from userdatamodel.user import User
+from werkzeug.local import LocalProxy
+
+from authutils.errors import AuthError
+from authutils.token.validate import set_current_token, validate_request
+
+
+def _set_current_user(**kwargs):
+    flask.g.user = CurrentUser(**kwargs)
+    set_current_token(flask.g.user.claims)
+    return flask.g.user
+
+
+# Proxy for the current user.
+#
+# Other modules importing authutils can import ``current_user`` from here,
+# which will use ``_get_or_set_current_user`` to look up the user.
+current_user = LocalProxy(_set_current_user)
```

Comment:
I don't quite understand how this works... where is kwargs passed to _set_current_user ?

Reply:
with this it won't pass anything—only default kwargs